wagtail/wagtailadmin/static_src/wagtailadmin/js/modal-workflow.js

```diff
@@ -51,9 +51,15 @@ function ModalWorkflow(opts) {
     };
 
     self.loadResponseText = function(responseText) {
-        var response = eval('(' + responseText + ')');
+        var response;
 
-        self.loadBody(response);
+        try {
+            response = JSON.parse( responseText );
+        } catch ( error ) {
+            window.location = window.location.host;
```

Comment:
if the session has timed out, how would they be able to save their work without logging in again?